tests/test_catalog.py

```diff
@@ -1218,6 +1218,26 @@ def test_get_all_collections(self) -> None:
         self.assertGreater(len(all_collections), 0)
         self.assertTrue(all(isinstance(c, pystac.Collection) for c in all_collections))
 
+    def test_get_single_links_media_type(self) -> None:
+        catalog = TestCases.test_case_9()
```

Comment:
```suggestion
        catalog = TestCases.test_case_1()

        catalog.links.append(
            pystac.Link(rel="search", target="./search.html", media_type="text/html")
        )
        catalog.links.append(
            pystac.Link(
                rel="search", target="./search.json", media_type="application/geo+json"
            )
        )
```
See comment below about manually adding the links instead of creating a new test case.